### PR TITLE
fix SectionEditorWebViewMessagingController retain cycle

### PR DESCRIPTION
**Pull request by tonisevener**
This wraps up https://phabricator.wikimedia.org/T251548.

**Testing steps:**

1. Open Instruments
2. Filter allocations by `SectionEditorWebViewMessagingController`.
3. Edit an article, complete flow until you're back on the article screen and it has reloaded.
4. 6.6 would still have a `SectionEditorWebViewMessagingController` instance hanging around in memory, this PR fixes that.

### CODE REVIEW

File: Wikipedia/Code/SectionEditorViewController.swift [hunk 1]
```diff
@@ -136,6 +147,31 @@ class SectionEditorViewController: ViewController {
...
+        for handler in scriptMessageHandlers {
+            userContentController.removeScriptMessageHandler(forName: handler.name)
+        }
+        
+        scriptMessageHandlers.removeAll()
+    }
+    
+    private func addScriptMessageHandlers(to contentController: WKUserContentController) {
+        
+        guard !scriptMessageHandlers.isEmpty else {
```
Comment (mcleinman) [inline Wikipedia/Code/SectionEditorViewController.swift:166]: Not a bug, but is this necessary?  If the array is empty, we just won't iterate over anything, no?

File: Wikipedia/Code/SectionEditorViewController.swift [hunk 2]
```diff
@@ -136,6 +147,31 @@ class SectionEditorViewController: ViewController {
         navigationController?.setNavigationBarHidden(false, animated: false)
     }
     
+    private func removeScriptMessageHandlers() {
+        
+        guard let userContentController = webView?.configuration.userContentController,
+            !scriptMessageHandlers.isEmpty else {
```
Comment (mcleinman) [inline Wikipedia/Code/SectionEditorViewController.swift:153]: same as below, not sure if we need this check - but not a problem to keep it

**Comment by mcleinman** [comment]
Will let you respond before I merge it - happy to merge it as-is.

File: Wikipedia/Code/SectionEditorViewController.swift (hunk 1 above)
Comment (tonisevener) [reply Wikipedia/Code/SectionEditorViewController.swift]: Good point! I removed this.

File: Wikipedia/Code/SectionEditorViewController.swift (hunk 2 above)
Comment (tonisevener) [reply Wikipedia/Code/SectionEditorViewController.swift]: Fixed.

**Pull request merged**